pkg/policy/rule.go

```diff
@@ -633,6 +633,14 @@ func (r *rule) matchesSubject(securityIdentity *identity.Identity) bool {
 	return r.subjectSelector.Selects(versioned.Latest(), securityIdentity.ID)
 }
 
+func (r *rule) getSubjects() []identity.NumericIdentity {
+	if r.NodeSelector.LabelSelector != nil {
+		return []identity.NumericIdentity{identity.ReservedIdentityHost}
+	}
+
+	return r.subjectSelector.GetSelections(versioned.Latest())
```

Comment:
Good question. Because of the ordering and locking here, we don't need to care about any sort of versioning, but it's interesting why.

We are interested in an identity being concurrently allocated while we are computing the set of affected selectors. The risk is that, somehow, an endpoint is created with an old policy version when it changes identities.

The key observation is that the repository is locked when updating or when calculating policy.

Because the repository is locked while we update rules and compute affected identities, it means that the endpoint's policy calculation is guaranteed to be entirely before or after applying the rule update. If it is before, then we know its identity is already in the selector cache and thus will be in the set of affected identities. If the endpoint calculates policy after the update is applied, then it will already have the new rule.

The final bit is that endpoints always recalculate policy if their identity changes.

So we don't need any special coordination between policy calculation and endpoint reading. We already guarantee that endpoint's security identity is in the selector cache prior to calculating policy, and the policy lock prevents changes while the endpoint is itself reading.